commit c506466fde8734115e1381fa802248fd0f7d5878 (refs/changes/37/4537/2)
Author: Jordan Paige Hendricks <jordan.hendricks@joyent.com>
Date:   2018-07-19T23:01:16+00:00 (1 year, 3 months ago)
    
    joyent/node-fast#20 allow specifying ports used for fastbench CLI

diff --git a/bin/fastbench b/bin/fastbench
index 644fedf..55a7b81 100755
--- a/bin/fastbench
+++ b/bin/fastbench
@@ -76,6 +76,8 @@ function fbUsageMessage()
 	    '    -i | --interval NSECONDS     Report every NSECONDS seconds.',
 	    '    -n | --nrequests NREQUESTS   Stop after completing ' +
 		'NREQUESTS requests.',
+	    '    -p | --artedi-port PORT       Use PORT for metrics endpoint.',
+	    '    -k | --kang-port PORT         Use PORT for kang endpoint.',
 	    ''
 	].join('\n');
 
@@ -106,7 +108,9 @@ function main()
 	});
 
 	parser = new mod_getopt.BasicParser(
-	    'c:(count)d:(duration)i:(interval)n:(nrequests)', process.argv);
+	    'c:(count)d:(duration)i:(interval)n:(nrequests)' +
+		'p:(artedi-port)k:(kang-port)',
+	     process.argv);
 	while ((option = parser.getopt()) !== undefined) {
 		switch (option.option) {
 		case 'c':
@@ -126,6 +130,14 @@ function main()
 			config.fbNRequests = parseOptInt(option, 1);
 			break;
 
+		case 'p':
+			config.fbArtediPort = parseOptInt(option, 1);
+			break;
+
+		case 'k':
+			config.fbKangPort = parseOptInt(option, 1);
+			break;
+
 		default:
 			/* error message already emitted */
 			mod_assertplus.equal('?', option.option);
